torrent/tracker.go

```diff
@@ -206,19 +208,24 @@ func (self *Tracker) serveFile(w http.ResponseWriter, r *http.Request, file *Fil
 		// TODO: This could run infinitely in a case where the file is requested and deleted or
 		// replaced, so we keep checking a structure that never will get filled in since it's no
 		// longer active.
+		file.Lock.Lock()
```

Comment:
So, `serveFile` is the hot path. When we push out a file and have 20,000 people start requesting it, this will force us to serialize that action. Do we think this is OK?

What do you think about using a RWLock instead, so that we can at least parallelize the serving flow?